terraform-post-packer/main.tf

```diff
@@ -24,34 +24,86 @@ locals {
   account_name_regex  = format("^env[[:digit:]]+ \\(%s\\)$", local.images_account_type)
 }
 
-# The IDs of all cisagov/skeleton-packer AMIs
-data "aws_ami_ids" "historical_amis" {
+# The IDs of all ARM64 cisagov/skeleton-packer AMIs
+data "aws_ami_ids" "historical_amis_arm64" {
+  owners = [data.aws_caller_identity.images.account_id]
+
+  filter {
+    name   = "architecture"
+    values = ["arm64"]
+  }
+
   filter {
-    name = "name"
-    values = [
-      "example-hvm-*-x86_64-ebs",
-    ]
+    name   = "name"
+    values = ["example-hvm-*-arm64-ebs"]
+  }
+
+  filter {
+    name   = "root-device-type"
+    values = ["ebs"]
   }
 
   filter {
     name   = "virtualization-type"
     values = ["hvm"]
   }
+}
+
+# Assign launch permissions to the ARM64 AMIs
+module "ami_launch_permission_arm64" {
+  # Really we only want the var.recent_ami_count most recent AMIs, but
+  # we have to cover the case where there are fewer than that many
+  # AMIs in existence.  Hence the min()/length() tomfoolery.
+  for_each = toset(slice(data.aws_ami_ids.historical_amis_arm64.ids, 0, min(var.recent_ami_count, length(data.aws_ami_ids.historical_amis_arm64.ids))))
+
+  source = "github.com/cisagov/ami-launch-permission-tf-module"
+
+  providers = {
+    aws        = aws
+    aws.master = aws.master
+  }
+
+  account_name_regex   = local.account_name_regex
+  ami_id               = each.value
+  extraorg_account_ids = var.extraorg_account_ids
+}
+
+# The IDs of all x86-64 cisagov/skeleton-packer AMIs
+data "aws_ami_ids" "historical_amis_x86_64" {
+  owners = [data.aws_caller_identity.images.account_id]
+
+  filter {
+    name   = "architecture"
+    values = ["x86_64"]
+  }
+
+  filter {
+    name   = "name"
+    values = ["example-hvm-*-x86_64-ebs"]
+  }
 
   filter {
     name   = "root-device-type"
     values = ["ebs"]
   }
 
-  owners = [data.aws_caller_identity.images.account_id]
+  filter {
+    name   = "virtualization-type"
+    values = ["hvm"]
+  }
+}
+
+moved {
```

Comment:
We should create an issue to remove this in a future (if not the next) kraken.

Reply:
Why?  That goes against [the advice from Terraform](https://developer.hashicorp.com/terraform/language/modules/develop/refactoring#removing-moved-blocks).

Reply:
That advice is based on upgrade paths for module users.

> We strongly recommend that you retain all historical `moved` blocks from earlier versions of your modules to preserve the upgrade path for users of any previous version.

Since these are our own root modules I don't think we need to worry about that scenario in the same way. I am not suggesting we remove them posthaste, but I do not see the harm in removing them at some point in the future given the nature of these projects.

Reply:
I'm not convinced, at least not yet. 

1. The moved blocks do no harm if left in. 
2. We don't know who else is using this repo. It's a public repo. Hence it's good practice to leave them in.

Reply:
Please see #369 and commit [bd80fc7](https://github.com/cisagov/skeleton-packer/pull/355/commits/bd80fc7b435e7ddc09b009bba605fc7452a03c98).

Reply:
Just to finish this off around what I mentioned in-person:

1. Though they do no harm they feel like unnecessary clutter to me as time goes on.
2. If we deprecate and remove them as part of the versioning of this project there should be no harm.
3. The Terraform configuration in question is only used to assign permissions to built AMIs and is not included in any other Terraform configuration, Therefore, the risk of blocking an upgrade path for consumers of this configuration are slim (in my opinion).